bugbug/github.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import logging
+from typing import Iterator, NewType, Tuple
+
+import requests
+from ratelimit import limits, sleep_and_retry
+
+from bugbug import db
+from bugbug.utils import get_secret
+
+logger = logging.getLogger(__name__)
+
+GITHUB_ISSUES_DB = "data/github_issues.json"
+db.register(
+    GITHUB_ISSUES_DB,
+    "https://community-tc.services.mozilla.com/api/index/v1/task/project.bugbug.data_github_issues.latest/artifacts/public/github_issues.json.zst",
+    1,
+)
+
+IssueDict = NewType("IssueDict", dict)
+
+GITHUB_TOKEN = get_secret("GITHUB_TOKEN")
+PER_PAGE = 100
+# Rate limit period in seconds
+RATE_LIMIT_PERIOD = 900
+
+
+def get_issues() -> Iterator[IssueDict]:
+    yield from db.read(GITHUB_ISSUES_DB)
+
+
+@sleep_and_retry
+@limits(calls=1200, period=RATE_LIMIT_PERIOD)
```

Comment:
Isn't the rate limit higher than this?

Note: it'd be nice to use the API to get the rate limit, so we don't wait needlessly (https://docs.github.com/en/rest/reference/rate-limit), but we can do that in a follow-up.

Reply:
Yeah, it's a bit higher (5000/hour so 1250 per 15 minutes). I went with 1200 per 15 minutes, to make sure we never hit the github rate limit exception, and after these 1200 the script sleeps for 15 minutes. 

I've tried another approach, where the script just fetches issues until github throws an exception, then makes a request to rate-limit api to fetch time until the reset and waits for it to reset. In this case wait time can be quite long, but we get more data between waits. 

But yeah, I can experiment and improve this in a follow-up :)